app/Http/Controllers/ChatUserController.php

```diff
@@ -39,4 +40,33 @@ public function store(Request $request, $reciever_id)
 
         return back();
     }
+
+    public function edit(ChatUser $chat)
+    {
+        if (Auth::User()->id != $chat->sender_id) {
+            return redirect('/404');
```

Comment:
Isn't tested and is 404 really the best place to redirect user ? I would redirect to back with flash error, or juste back or home.